### PR TITLE
Deprecate spi params

### LINKED ISSUE #2: Deprecate SPI params bcm_DC, bcm_RST in favour of gpio_DC, gpio_RST
See https://github.com/rm-hull/luma.core/blob/master/luma/core/serial.py#L96

*bcm* is shorthand for 'broadcom' pin assignment, and is specific to RPi implementations, whereas *gpio* is a more general term, and should be used if ever a generic GPIO library replaces RPi.GPIO

### PR DESCRIPTION
**Pull request by thijstriemstra**
fixes #2

### CODE REVIEW

**Comment by thijstriemstra** [comment]
work in progress..

**Comment by thijstriemstra** [comment]
Enabling warnings displays a lot of resource warnings: https://travis-ci.org/rm-hull/luma.core/jobs/210702792#L472. I'll fix those as well.

**Comment by thijstriemstra** [comment]
@rm-hull the resource warnings coming from `test_spritesheet` could be considered bugs because the luma code doesn't close the image for you. But I'm not sure where to close the image in the `spritesheet` class..

```
WW0 in tests/test_spritesheet.py:42 the following warning was recorded:
 .tox/py35/lib/python3.5/site-packages/PIL/Image.py:1028: ResourceWarning: unclosed file <_io.BufferedReader name='tests/reference/runner.png'>
  self.load()
```

File: tests/test_serial.py [hunk 1]
```diff
@@ -138,14 +138,42 @@ def verify_spi_init(port, device, bus_speed_hz=8000000, dc=24, rst=25):
...
     ])
 
 
+def test_spi_init_params_deprecated():
+    port = 5
+    device = 2
+    bus_speed = 16000000
+    dc = 80
+    rst = 90
+    msg1 = 'bcm_DC argument is deprecated in favor of gpio_DC and will be removed in 1.0.0'
```
Comment (thijstriemstra) [inline tests/test_serial.py]: @rm-hull this is an arbitrary version number, what would be a good version to drop this?

Comment (rm-hull) [inline tests/test_serial.py]: 1.0.0 is probably the right version ... as long as we dont forget!

**Comment by rm-hull** [comment]
Think you're right about the spritesheet - maybe adding a `__del__()` destructor method to the spritesheet class that calls `self.image.close()` ?

**Comment by thijstriemstra** [comment]
@rm-hull spritesheet probably needs a `close` method. This can be called manually and if spritesheet would support context manager usage, `close` could be called there.

File: tests/test_serial.py (hunk 1 above)
Comment (thijstriemstra) [inline tests/test_serial.py]: I'll open a ticket for it.

File: luma/core/sprite_system.py [hunk 2]
```diff
@@ -122,6 +124,9 @@ def __len__(self):
         """
         return self.frames.count
 
+    def close(self):
+        self._img.close()
+
```
Comment (thijstriemstra) [inline luma/core/sprite_system.py]: unfortunately self.image.close() didn't work so added a new private var that can be closed.

**Comment by thijstriemstra** [comment]
Should be ready for review.

**Comment by rm-hull** [comment]
Just reading the docs for https://pillow.readthedocs.io/en/4.0.x/reference/Image.html#PIL.Image.Image.load

This implies as soon as an image is accessed, `load` is called and the underlying file pointer is closed. Are we unnecessarily complicating things? Are the warnings actually just false alarms?

**Comment by thijstriemstra** [comment]
Pillow seems buggy with closing them. Pillow should also support the `with` keyword but it's not working. Also see http://stackoverflow.com/a/38542329

File: luma/core/sprite_system.py [hunk 3]
```diff
@@ -66,7 +67,8 @@ class spritesheet(object):
     Loosely based on http://www.createjs.com/docs/easeljs/classes/SpriteSheet.html
     """
     def __init__(self, image, frames, animations):
-        self.image = Image.open(image)
+        self._img = open(image, 'rb')
+        self.image = Image.open(self._img)
```
Comment (rm-hull) [inline luma/core/sprite_system.py]: this works (tested), and doesn't rely on the consumer to call close:

```python
    def __init__(self, image, frames, animations):
        with open(image, "rb") as fp:
            self.image = Image.open(fp)
            self.image.load()

        ...
```

presumably it wont trigger the warning

Comment (thijstriemstra) [inline luma/core/sprite_system.py]: thanks that worked. updated PR.

**Comment by rm-hull** [comment]
Can you squash & merge when everything goes green pls ?

**Comment by thijstriemstra** [comment]
I guess the examples now need to updated, working on PR.

**Pull request merged**